src/app/choice.rs

```diff
@@ -406,8 +406,21 @@ impl<G: Group, S: ProveSum<G>> EncryptedChoice<G, S> {
     pub fn sum_proof(&self) -> &S::Proof {
         &self.sum_proof
     }
+
+
+    /// Returns a new encrypted choice
+    pub fn new_encrypted_choice(choices: Vec<Ciphertext<G>>, range_proof: RingProof<G>, sum_proof: S::Proof) -> Self {
```

Comment:
Bikeshedding: I'd prefer that this method would be called something like `from_parts`. Also, the docs should specify that if parts do not represent a valid encrypted choice, it cannot be used successfully (i.e., the caller is responsible for the args validation).